public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -620,7 +620,7 @@ export function getOriginalMetricName(labelData: { [key: string]: string }) {
 }
 
 function mergeHeatmapFrames(frames: DataFrame[]): DataFrame[] {
-  if (frames.length === 0) {
+  if (frames.length === 0 || (frames.length === 1 && frames[0].fields.length === 0)) {
```

Comment:
maybe this? (untested)

```suggestion
  if (frames.length === 0 || (frames.length === 1 && frames[0].length === 0)) {
```

Reply:
Both of these proposed changes are resolving the bug locally for me

Reply:
Same, I went ahead and committed his suggestion!